### ISSUE TITLE
Compute shaders only write to the first layer in a 3D texture in Direct3D11.

### ISSUE THREAD

**Issue by bensmooth** [mellinoe/veldrid#351]
It seems that compute shaders with the D3D11 backend are unable to write to 3D textures past z=0.  The same code running with the Vulkan backend is able to write to the entire depth of the texture.  

With RenderDoc, I have verified that the output texture only gets populated for z=0 when using the D3D11 backend.  All of the other layers are filled with (0, 0, 0, 0).  Am I missing something, or is this a deficiency in the way Veldrid is interfacing with D3D11?  

Here is a ~200 line example project for reproduction:
https://github.com/bensmooth/Veldrid3dComputeShader
I make a 32x32x32 3D texture and run a dumb compute shader on it that just fills the texture with a value from a uniform.  Then, I copy each layer of the texture back and make sure it actually got filled properly.  With the D3D backend, only the first layer is properly filled.

Running runD3D.bat produces results I would call incorrect.  Running runVulkan.bat produces the results I would expect.  

Anyone have any hints as to where I can start digging around in Veldrid's D3D implementation to figure this out?

**Comment by bensmooth** [comment]
After some finagling, I've added my example as a test case (ComputeShader3dTexture) to the library here:
https://github.com/bensmooth/veldridFix3dComputeD3D

The D3D11 backend fails to fill all of the layers of the 3D texture.  The Vulkan backend successfully fills all of the layers, as expected.

For whatever reason, something is wrong with reading back the staging texture in OpenGL.  I have verified with RenderDoc that OpenGL does fill in the 3D texture with the right values with the compute shader, but when we map the staging texture it is all zeros.
